src/screens/ScanQrCodeScreen.tsx

```diff
@@ -16,16 +16,20 @@ interface Props {
   route: RouteProp<MainCardStackNavigatorParams, Route.ScanQrCode>;
 }
 export default class ScanQrCodeScreen extends React.PureComponent<Props> {
-  static navigationOptions = {
-    header: null,
+  state = {
+    scanned: false,
   };
-
-  onBarCodeScanned = async (scannedQr: any) => {
-    const { route, navigation } = this.props;
+  goBack = () => this.props.navigation.goBack();
+  onBarCodeScanned = (scannedQr: any) => {
+    if (this.state.scanned) {
+      return;
+    }
+    this.setState({ scanned: true });
```

Comment:
I'm wondering if it will always work because setState is an async function so, it is possible that the scanned won't be set to true before next execution of onBarCodeScanned